src/Components/EventRegistration/index.jsx

```diff
@@ -0,0 +1,308 @@
+      });
+    }
+  };
+  const removeSlot = (ind) => {
+    if (teamMemberDetails.length > event.minMembers) {
+      setErrorDetails((prevDets) => {
+        const newArr = [...prevDets];
+        newArr.splice(ind, 1);
+        return newArr;
+      });
+      setTeamMemberDetails((prevDets) => {
+        const newArr = [...prevDets];
+        newArr.splice(ind, 1);
+        return newArr;
+      });
+    }
+  };
+  const [teamMemberDetails, setTeamMemberDetails] = useState(
+    Array.from({ length: event.minMembers }).fill("")
+  );
+  const [errorDetails, setErrorDetails] = useState(
+    Array.from({ length: event.minMembers }).fill("")
+  );
+ 
+  useEffect(() => {
+    if(!alreadyRegistered){
+      setTeamMemberDetails((prevDets) => {
+        const newArr = [...prevDets];
+        newArr[0] = userDets?.pecfestId;
+        return newArr;
+      });
+    }
+    else{
+      (async ()=>{
+        const teamData = await getDocById('event-teams', registeredTeamId);
+        console.log('teamData', teamData);
+        setTeamMemberDetails(teamData?.teamMembers);
+      })();
+    }
+   
+  }, []);
+  const validateUnique = () => {
+    const teamSet = new Set();
+    let finalValid = true;
+    teamMemberDetails.forEach((teamMember, ind) => {
+      if (!teamMember) {
+        return;
+      }
+      if (teamSet.has(teamMember)) {
+        finalValid = false;
+        setErrorDetails((prevDets) => {
+          const newArr = [...prevDets];
+          newArr[ind] = "Team members must be unique";
+          return newArr;
+        });
+      } else {
+        teamSet.add(teamMember);
+      }
+    });
+    return finalValid;
+  };
+  const validateForm = async () => {
+    let errorOccured = false;
+    setErrorDetails(Array.from({ length: teamMemberDetails.length }).fill(""));
+    if (!validateUnique()) {
+      errorOccured = true;
+    }
+    const usersData = await getUsersInPECIdArray(teamMemberDetails);
+    if (!validatePecId(usersData)) {
+      errorOccured = true;
+    }
+    //   //
+    if (!errorOccured) {
+      if (event.isTeamEvent) {
+        // generate team id
+        const teamDets = await createDoc("event-teams", {
+          teamMembers: teamMemberDetails,
+        });
+        // save this in every user and inside event
+        usersData.forEach((user) => {
+          const key = "registeredEvents";
+          let eventsList = user[key];
+
+          if (!eventsList) {
+            eventsList = [];
+          }
+          try {
+            const payloadData = {
+              ...user,
+              [key]: [
+                ...eventsList,
+                { eventId: event.id, prelimLink, teamId: teamDets.id },
+              ],
+            };
+            updateDoc("users", user.id, payloadData);
+          } catch (err) {
+            console.log(err);
+            toast.error("Something went wrong");
+          }
+        });
+        if (!alreadyRegistered) {
+          onRegister(teamDets.id);
```

Comment:
this function is again updating the users collection, which is already updated above

Reply:
upar its for the team, this one updates the event ki